ext/opentelemetry-ext-django/src/opentelemetry/ext/django/middleware.py

```diff
@@ -0,0 +1,110 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from logging import getLogger
+
+from django.utils.deprecation import MiddlewareMixin
+
+from opentelemetry.context import attach, detach
+from opentelemetry.ext.django.version import __version__
+from opentelemetry.ext.wsgi import (
+    add_response_attributes,
+    collect_request_attributes,
+    get_header_from_environ,
+)
+from opentelemetry.propagators import extract
+from opentelemetry.trace import SpanKind, get_tracer
+
+_logger = getLogger(__name__)
+
+
+class DjangoMiddleware(MiddlewareMixin):
```

Comment:
Should this be private to emphasize we do not expect users to instantiate these?

Reply:
Fixed